cmd/root_linux_test.go

```diff
@@ -71,3 +75,77 @@ func TestNewCommandArgumentsOnLinux(t *testing.T) {
 		})
 	}
 }
+
+func TestSdNotifyOnLinux(t *testing.T) {
+	tcs := []struct {
+		desc          string
+		proxyMustFail bool
+		notifyState   string
+	}{
+		{
+			desc:          "System with systemd Type=notify and proxy started successfully",
+			proxyMustFail: false,
+			notifyState:   daemon.SdNotifyReady,
+		},
+		{
+			desc:          "System with systemd Type=notify and proxy failed to start",
+			proxyMustFail: true,
+			notifyState:   daemon.SdNotifyStopping,
+		},
+	}
+
+	// Create a temp dir for the socket file.
+	testDir, err := os.MkdirTemp("/tmp/", "test-")
+	if err != nil {
+		t.Fatalf("Fail to create the temp dir: %v", err)
+	}
+	defer os.RemoveAll(testDir)
+
+	//Set up the socket stream to listen for notifications.
+	socketAddr := testDir + "/notify-socket.sock"
+	laddr := net.UnixAddr{
```

Comment:
minor nit: `&net.UnixAddr{}` is more common in my experience, rather than taking the address below.

Reply:
updated